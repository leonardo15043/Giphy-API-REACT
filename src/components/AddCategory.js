import React, { Fragment,useState } from 'react'
import PropTypes from 'prop-types'

function AddCategory(props) {

    const [ inputValue, setInputValue ] = useState('Hola Mundo')

    const handleInputChange = (e)=>{
        setInputValue( e.target.value );
    }

    return (
        <Fragment>
            <h2>App Categories</h2>
            <input type="text" value={ inputValue } onChange={ handleInputChange } />
        </Fragment>
    )
}

AddCategory.propTypes = {

}

export default AddCategory



