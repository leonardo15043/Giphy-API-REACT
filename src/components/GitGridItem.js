import React from 'react'

export const GitGridItem = ( { id, title, url } ) => {

    return (
        <div className="card animate__animated animate__fadeInLeft">
            <img src={ url } alt={ title } />
            <p>{ title}</p>
        </div>
    )
}
