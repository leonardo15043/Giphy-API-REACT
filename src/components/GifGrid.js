import React, { Fragment, useEffect, useState } from 'react';
import { GitGridItem } from './GitGridItem';

export const GifGrid = ({ category }) => {

    const [images, setimages] = useState([]);

    useEffect( ()=>{
        getGifs();
    },[])

    const getGifs = async() =>{
        const url = 'https://api.giphy.com/v1/gifs/search?q=One+Punch&limit=10&api_key=48Qzvn6bYuINFc6tnXrL0MfuU7Njc9zf';
        const resp = await fetch( url );
        const { data } = await resp.json();

        const gifs = data.map( img => {
            return {
                id : img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })

        setimages(gifs);
    }

    return ( 
        <Fragment>
        <h3> { category } </h3>
        <div className="card-grid">
            
            <ol>
                {
                    images.map( img =>(
                        <GitGridItem key={ img.id }  { ...img } />
                    ))
                } 
            </ol>
        </div>
        </Fragment>

    )

}
