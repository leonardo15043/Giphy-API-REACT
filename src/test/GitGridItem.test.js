import { shallow } from "enzyme"
import { GitGridItem } from "../components/GitGridItem"

describe('Pruebas en GitGridItem', () => {
    test('Cargar correctamente el componente', () => {

        const image = { 
            id: 12,
            title: 'Bored One',
            url : 'https://media4.giphy.com/media/6IkjQmpaRwIabJ2G3C/giphy-downsized-medium.gif?cid=cd622c90mqvu6s36rmyko0ies0n0xgn9t26d0wqtlmkvdpzw&rid=giphy-downsized-medium.gif'
        }

        const wapper = shallow(<GitGridItem { ...image } />);

        expect( wapper ).toMatchSnapshot();
        
    })  
})
