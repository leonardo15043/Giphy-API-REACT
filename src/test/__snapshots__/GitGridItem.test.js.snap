// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GitGridItem Cargar correctamente el componente 1`] = `
<div
  className="card animate__animated animate__fadeInLeft"
>
  <img
    alt="Bored One"
    src="https://media4.giphy.com/media/6IkjQmpaRwIabJ2G3C/giphy-downsized-medium.gif?cid=cd622c90mqvu6s36rmyko0ies0n0xgn9t26d0wqtlmkvdpzw&rid=giphy-downsized-medium.gif"
  />
  <p>
    Bored One
  </p>
</div>
`;
